boards/acd52832/src/main.rs

```diff
@@ -584,7 +584,7 @@ pub unsafe fn main() {
         static _eappmem: u8;
     }
 
-    kernel::process::load_processes(
+    kernel::process::load_and_check_processes(
```

Comment:
It is a little strange to me that whether this function actually checks the processes just depends on the option passed to the creation of `board_kernel`.

Currently this PR seems to increase kernel code size by a baseline of about 2.5 kB for all boards -- that is a pretty large cost for boards that do not need it, and I think we should investigate how to reduce that

Reply:
Agreed -- the cost when you are not using a verifier should be small. I will guess it's related to Leon's point about generics, and whether the state machine is included.

Reply:
> It is a little strange to me that whether this function actually checks the processes just depends on the option passed to the creation of `board_kernel`.

What else should it depend on? The goal is to encapsulate the entire functionality of getting userspace processes to run. However, if a board doesn't want to check processes (e.g., to reduce code size), it doesn't have to; a `None` for the checker defaults to no checking. But I agree that it would be better if not having a checker means the entire state machine (and code) is elided, which a `None` seems to not be doing. I'll look into the generic type approach.